src/Model/ROI.py

```diff
@@ -926,3 +1009,133 @@ def renumber_roi_number(sequence):
             item.add_new(Tag("ReferencedROINumber"), "IS", str(roi_number))
         roi_number += 1
     return sequence
+
+
+def roi_to_geometry(dict_rois_contours):
+    """
+    Convert ROI contour data in each image slice to a geometry object
+    :param dict_rois_contours: A dictionary with key-value pair
+    {slice-uid: contour sequence}
+    :return: A dictionary with key-value pair {slice-uid: Geometry object}
+    """
+    dict_geometry = {}
+
+    for slice_uid, contour_sequence in dict_rois_contours.items():
+        # convert contour data to polygon omitting data with less than 3 points
+        polygon_list = [Polygon(contour_data)
+                        for contour_data in contour_sequence
+                        if len(contour_data) >= 3]
+        result_geometry = make_valid(MultiPolygon(polygon_list))
+        dict_geometry[slice_uid] = result_geometry
+
+    return dict_geometry
+
+
+geometry_manipulation = {
```

Comment:
If these are meant to be dict of types of manipulations - it might be better to place these at the top of the file. Otherwise, it seems to get lost in the huge chunk of code and considering it's not exactly quite a defined function but rather a data structure of macros, it shouldn't necessarily be placed in the middle of the haystack.

Reply:
Done